src/index/IndexImpl.cpp

```diff
@@ -543,27 +545,18 @@ IndexImpl::createPermutationPairImpl(const string& fileName1,
     lastLhs = triple[c1];
   }
   if (from < totalNumTriples) {
-    writer1.addRelation(currentRel.value(), buffer, distinctCol1);
-    writeSwitchedRel(&writer2, currentRel.value(), &buffer);
+    addCurrentRelation();
   }
 
-  writer1.finish();
-  writer2.finish();
-  for (auto& md : writer1.getFinishedMetaData()) {
-    metaData1.add(md);
-  }
-  for (auto& md : writer2.getFinishedMetaData()) {
-    metaData2.add(md);
-  }
-  metaData1.blockData() = writer1.getFinishedBlocks();
-  metaData2.blockData() = writer2.getFinishedBlocks();
+  metaData1.blockData() = std::move(writer1).getFinishedBlocks();
+  metaData2.blockData() = std::move(writer2).getFinishedBlocks();
 
   return std::make_pair(std::move(metaData1), std::move(metaData2));
 }
 
 // __________________________________________________________________________
-void IndexImpl::writeSwitchedRel(CompressedRelationWriter* out, Id currentRel,
-                                 BufferedIdTable* bufPtr) {
+CompressedRelationMetadata IndexImpl::writeSwitchedRel(
+    CompressedRelationWriter* out, Id currentRel, BufferedIdTable* bufPtr) {
```

Comment:
Sonarcloud says that this function can be made `const`, but I think it can even be `static` (Sonar doesn't check this, but maybe the `clang-tidy` in your CLION does.
(not your fault, but we improve the touched code as we go)

Reply:
I saw that the suggestion by Sonarcloud and clang-tidy but they didn't agree. But I guess `static` could be seen as stronger than `const` and be used instead of `const`.

Reply:
Yes.
Sonarcloud doesn't have the check for "can be made `static`", so it "only" suggests `const` in these cases. Maybe we should also integrate another static analysis tool into the CI pipeline to make reviewing even simpler:)

Reply:
With multiple different code quality tools it might be worth having a look at annotations. They can be added prints in specific format (see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message) and would then appear in the GitHub interface: 
![image](https://user-images.githubusercontent.com/14220769/233658821-418a720e-6d6c-4db7-be13-e3f998263b97.png)

Reply:
Maybe that could be an option,  another would be to manually add (certain) clang-tidy warnings to Sonarcloud.
The problem with annotations directly on GitHub is that they tend to clutter the code. I generally don't mind the extra click to Sonarcloud.